api/src/main/java/org/openmrs/module/santedb/mpiclient/api/impl/FhirMpiClientServiceImpl.java

```diff
@@ -222,47 +257,61 @@ else if("basic".equals(this.m_configuration.getAuthenticationMode())) {
-//            } else
-                query = query.where(org.hl7.fhir.r4.model.Patient.IDENTIFIER.exactly()
-                        .identifier(patientIdentifier.getIdentifier()));
+        if (patientIdentifiers != null) {
+            for (PatientIdentifier patientIdentifier : patientIdentifiers) {
+                if (patientIdentifier.getIdentifierType() != null) {
+
+                    HashMap<String, String> localPatientIdentifierTypeMap = this.m_configuration.getLocalPatientIdentifierTypeMap();
+                    String authority = localPatientIdentifierTypeMap
+                            .get(patientIdentifier.getIdentifierType().getName());
+                    if (authority == null)
+                        throw new MpiClientException(
+                                String.format("Identity domain %s doesn't have an equivalent in the MPI configuration",
+                                        patientIdentifier.getIdentifierType().getName()));
+                    query = query.where(org.hl7.fhir.r4.model.Patient.IDENTIFIER.exactly()
+                            .systemAndIdentifier(patientIdentifier.getIdentifier(), authority));
+                } else {
+//                    Skip the search by the identifier
+                }
+            }
+        }
+        String localBiometricSubjectId = (String) otherDataPoints.get("localBiometricSubjectId");
+        String nationalBiometricSubjectId = (String) otherDataPoints.get("nationalBiometricSubjectId");
+        String phoneNumber = (String) otherDataPoints.get("phoneNumber");
+
+        if (StringUtils.isNotBlank(localBiometricSubjectId)) {
+            query = query.where(org.hl7.fhir.r4.model.Patient.IDENTIFIER.exactly()
+                    .systemAndIdentifier(localBiometricSubjectId, "Biometrics Reference Code"));
+        }
+        if (StringUtils.isNotBlank(nationalBiometricSubjectId)) {
+            query = query.where(org.hl7.fhir.r4.model.Patient.IDENTIFIER.exactly()
+                    .systemAndIdentifier(nationalBiometricSubjectId, "Biometrics National Reference Code"));
+        }
 
+        if (StringUtils.isNotBlank(phoneNumber)) {
+            query = query.where(org.hl7.fhir.r4.model.Patient.PHONE.exactly().identifier(phoneNumber));
         }
 
+
         if (stateOrRegion != null && !stateOrRegion.isEmpty())
             query = query.where(org.hl7.fhir.r4.model.Patient.ADDRESS_STATE.contains().value(stateOrRegion));
         if (cityOrTownship != null && !cityOrTownship.isEmpty())
             query = query.where(org.hl7.fhir.r4.model.Patient.ADDRESS_CITY.contains().value(cityOrTownship));
 
-        // Send the message and construct the result set
-        try {
+        return query;
+    }
 
-            Bundle results = query.returnBundle(Bundle.class).execute();
+    @Override
+    public List<MpiPatient> searchPatient(String familyName, String givenName, Date dateOfBirth, boolean fuzzyDate, String gender,
```

Comment:
can we organize the two `searchPatient` functions together and add a small comment about their different uses?